web/session.py

```diff
@@ -309,12 +309,13 @@ def __delitem__(self, key):
             os.remove(path)
 
     def cleanup(self, timeout):
-        now = time.time()
-        for f in os.listdir(self.root):
-            path = self._get_path(f)
-            atime = os.stat(path).st_atime
-            if now - atime > timeout:
-                os.remove(path)
+        if os.path.isdir(self.root):
+            now = time.time()
+            for f in os.listdir(self.root):
+                path = self._get_path(f)
+                atime = os.stat(path).st_atime
+                if now - atime > timeout:
+                    shutil.rmtree(path)
```

Comment:
I suggest changing this to:

```
        if not os.path.isdir(self.root):
            return
        ...
```

It is hard to read when the most important part of the function is packed into another nested blocked.

Reply:
Fixed.